src/exo/LoopIR_scheduling.py

```diff
@@ -1848,12 +1849,16 @@ def idx_mode(self, access, orig):
 
     def map_s(self, s):
         if s is self.alloc_stmt:
-            # mark the point we want to lift this alloc-stmt to
-            n_up = min(self.n_lifts, len(self.ctrl_ctxt))
-            self.lift_site = self.ctrl_ctxt[-n_up]
+            if self.n_lifts > len(self.ctrl_ctxt):
+                raise SchedulingError(
+                    "specified lift level {self.n_lifts} "
+                    + "is higher than the number of loop "
+                    + "{len(self.ctrl_ctxt)}"
```

Comment:
Missing an `f` here for format string. Also `+` is not needed.

Reply:
Woops thanks!